src/views/virtualmachines/list/hooks/useVirtualMachineColumns.ts

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import {
+  K8sResourceCommon,
+  TableColumn,
+  useActiveColumns,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { sortable } from '@patternfly/react-table';
+
+export const columnsStringSortingSubField =
+  <T>(field: string, subField: string) =>
+  (data: Array<T>, direction: string) =>
+    data?.sort((a, b) =>
+      direction === 'asc'
+        ? a?.[field]?.[subField]?.localeCompare(b?.[field]?.[subField])
+        : b?.[field]?.[subField]?.localeCompare(a?.[field]?.[subField]),
+    );
+
+const useVirtualMachineColumns = () => {
+  const { t } = useKubevirtTranslation();
+
+  const columns: TableColumn<K8sResourceCommon>[] = React.useMemo(
+    () => [
+      {
+        title: t('Name'),
+        id: 'name',
+        transforms: [sortable],
+        sort: columnsStringSortingSubField<K8sResourceCommon>('metadata', 'name'),
```

Comment:
why do we need this complex function? @metalice 
why not just write
`sort: metadata.name`

Reply:
I don't know, does metadata.name works? @avivtur

Reply:
+1 changed, thanks @glekner @metalice